File: BennuLib/IO/PalPaletteEncoder.vb
Namespace Bennu.IO
    Public Class PalPaletteEncoder
        Inherits NativeEncoder(Of Palette)

        Protected Overrides ReadOnly Property Version As Byte = &H0

        Protected Overrides Sub WriteNativeFormat(palette As Palette, writer As NativeFormatWriter)
            writer.Write(palette)
            writer.WriteReservedPaletteGammaSection()
        End Sub

        Protected Overrides Function GetFileId(obj As Palette) As String
            Return "pal"
        End Function
    End Class
End Namespace

File: BennuLib/BennuLib.Core/IO/PalPaletteEncoder.cs
namespace Bennu.IO
{
	public class PalPaletteEncoder : NativeEncoder<Palette>
	{
        private const int version = 0x00;

		protected override byte GetLastHeaderByte(Palette palette) => version;

		protected override void WriteNativeFormatBody(Palette palette, NativeFormatWriter writer)
		{
			writer.Write(palette);
			writer.WriteReservedPaletteGammaSection();
		}

		protected override string GetFileId(Palette obj)
		{
			return "pal";
		}
	}
}
